src/coreclr/src/nativeaot/Runtime/EHHelpers.cpp

```diff
@@ -356,6 +356,73 @@ static UIntNative UnwindWriteBarrierToCaller(
     return adjustedFaultingIP;
 }
 
+EXTERN_C void* RhpInitialInterfaceDispatch;
+EXTERN_C void* RhpInterfaceDispatchAVLocation1;
+EXTERN_C void* RhpInterfaceDispatchAVLocation2;
+EXTERN_C void* RhpInterfaceDispatchAVLocation4;
+EXTERN_C void* RhpInterfaceDispatchAVLocation8;
+EXTERN_C void* RhpInterfaceDispatchAVLocation16;
+EXTERN_C void* RhpInterfaceDispatchAVLocation32;
+EXTERN_C void* RhpInterfaceDispatchAVLocation64;
+
+static bool InInterfaceDispatchHelper(UIntNative faultingIP)
+{
+#ifndef USE_PORTABLE_HELPERS
+    static UIntNative interfaceDispatchAVLocations[] =
+    {
+        (UIntNative)&RhpInitialInterfaceDispatch,
+        (UIntNative)&RhpInterfaceDispatchAVLocation1,
+        (UIntNative)&RhpInterfaceDispatchAVLocation2,
+        (UIntNative)&RhpInterfaceDispatchAVLocation4,
+        (UIntNative)&RhpInterfaceDispatchAVLocation8,
+        (UIntNative)&RhpInterfaceDispatchAVLocation16,
+        (UIntNative)&RhpInterfaceDispatchAVLocation32,
+        (UIntNative)&RhpInterfaceDispatchAVLocation64,
+    };
+
+    // compare the IP against the list of known possible AV locations in the interface dispatch helpers
+    for (size_t i = 0; i < sizeof(interfaceDispatchAVLocations) / sizeof(interfaceDispatchAVLocations[0]); i++)
+    {
+#if defined(HOST_AMD64) || defined(HOST_X86)
+        // Verify that the runtime is not linked with incremental linking enabled. Incremental linking
+        // wraps every method symbol with a jump stub that breaks the following check.
+        ASSERT(*(UInt8*)interfaceDispatchAVLocations[i] != 0xE9); // jmp XXXXXXXX
+#endif
+
+        if (interfaceDispatchAVLocations[i] == faultingIP)
+            return true;
+    }
+#endif // USE_PORTABLE_HELPERS
+
+    return false;
+}
+
+static UIntNative UnwindInterfaceDispatchToCaller(
```

Comment:
This is the exact same code as the write barriers unwinder. I could just rename that one to a common name and call it.

I just wasn't sure whether this code could be different for other arches since here we have an unknown number of parameters. It happens to not be a problem on x64. I don't know about x86.

Reply:
x86 is full of problems that none of the other platforms have. I think having the unwinder shared would be a good idea.

Reply:
(If we ever do x86 for completeness, we may want to do it using cdecl callconv to avoid all special casing.)